File: app/src/main/java/lee/jaebeom/WikiPortal/java/SettingPresenter.java
/*
package lee.jaebeom.WikiPortal.setting;

import android.content.SharedPreferences;

import lee.jaebeom.WikiPortal.wiki.WikiData;


public class SettingPresenter implements SettingContract.Presenter {
    private final SettingContract.View view;

    public SettingPresenter(SettingContract.View view) {
        this.view = view;
        view._presenter(this);
    }

    @Override
    public void start() {

    }

    @Override
    public void setUsed(SharedPreferences sharedPreferences, String key) {
        WikiData.wikiURL.get(key).setUse(sharedPreferences.getBoolean(key, false));
//        if (WikiData.wikiURL.get(key).isUse()){
//            WikiData.wikis++;
//        }else{
//            WikiData.wikis--;
//        }
    }

    @Override
    public boolean isConfirm() {
        if(WikiData.wikis > 0){
            return true;
        }else{
            return false;
        }

    }
}
*/

File: app/src/main/java/lee/jaebeom/WikiPortal/setting/SettingPresenter.kt
package lee.jaebeom.WikiPortal.setting

import android.content.SharedPreferences
import lee.jaebeom.WikiPortal.wiki.WikiData

/**
 * Created by leejaebeom on 2017. 11. 9..
 */
class SettingPresenter(view : SettingContract.View) : SettingContract.Presenter{
    init {
        view._presenter(this)
    }
    override fun start() {

    }

    override fun setUsed(preferences: SharedPreferences, key: String) {
        WikiData.wikiURL.get(key)?.isUse = preferences.getBoolean(key, false)
    }

    override fun isConfirm(): Boolean {
        return WikiData.wikis!! > 0
    }
}